src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets

```diff
@@ -145,13 +145,27 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <UsingTask TaskName="Microsoft.NET.Build.Tasks.ResolvePackageDependencies"
              AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+  <UsingTask TaskName="Microsoft.NET.Build.Tasks.CheckForTargetInAssetsFile"
+           AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
 
   <!-- The condition on this target causes it to be skipped during design-time builds if
         the restore operation hasn't run yet.  This is to avoid displaying an error in
         the Visual Studio error list when a project is created before NuGet restore has
         run and created the assets file. -->
   <Target Name="RunResolvePackageDependencies"
           Condition=" '$(DesignTimeBuild)' != 'true' Or Exists('$(ProjectAssetsFile)')">
+    
+    <!-- Verify that the assets file has a target for the right framework.  Otherwise, if we restored for the
+         wrong framework, we'd end up finding no references to pass to the compiler, and we'd get a ton of
+         compile errors.
+         
+         The RuntimeIdentifier isn't used to select references to pass to the compiler, so we don't include
+         it here. -->
+    <CheckForTargetInAssetsFile
+      AssetsFilePath="$(ProjectAssetsFile)"
+      TargetFrameworkMoniker="$(TargetFrameworkMoniker)"
+      RuntimeIdentifier="" />
```

Comment:
I think we should come up with another way to test than to expect that we can pass an invalid rid if we're only compiling. It's not a vaild use case and it would make sense for the check here to be more defensive.

Options:

1. A package that causes compilation to fail iff you use its rid-ful assets.
2. Use GetValues to check ReferencePath is as we expect heading in to Csc.